packages/chain-adapters/src/utxo/UtxoBaseAdapter.ts

```diff
@@ -479,15 +495,8 @@ export abstract class UtxoBaseAdapter<T extends UtxoChainId> implements IChainAd
     }
   }
 
-  async broadcastTransaction({
-    senderAddress,
-    receiverAddress,
-    hex,
-  }: BroadcastTransactionInput): Promise<string> {
-    await Promise.all([
-      validateAddress(senderAddress),
-      receiverAddress !== CONTRACT_INTERACTION && validateAddress(receiverAddress),
-    ])
+  async broadcastTransaction({ receiverAddress, hex }: BroadcastTransactionInput): Promise<string> {
+    await (receiverAddress !== CONTRACT_INTERACTION && validateAddress(receiverAddress))
```

Comment:
Keeps only the receive side check now - the naming is odd but this refers (or at least *should* refer to) the *final* recipient e.g for a cross-chain swap, possibly a receive address on a diff chain